demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,361 @@
+#define REQUEST_BODY_TEXT_LENGTH    ( sizeof( REQUEST_BODY_TEXT ) - 1 )
+
+/**
+ * @brief A buffer used to store request headers and reused after sending
+ * the request to store the response headers and body.
+ *
+ * User can also decide to have two separate buffers for request and response.
+ */
+static uint8_t userBuffer[ USER_BUFFER_LENGTH ] = { 0 };
+
+/**
+ * @brief The request body.
+ */
+static uint8_t requestBodyBuffer[] = REQUEST_BODY_TEXT;
+
+/**
+ * @brief Definition of the HTTP network context.
+ */
+struct HTTPNetworkContext
+{
+    int tcpSocket;
+};
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Establish a TCP connection to the given server.
+ *
+ * @param[in] pServer Host name of server.
+ * @param[in] port Server port.
+ *
+ * @return A file descriptor representing the TCP socket; -1 on failure.
+ */
+static int connectToServer( const char * pServer,
+                            uint16_t port )
+{
+    int status, tcpSocket = -1;
+    struct addrinfo * pListHead = NULL, * pIndex;
+    struct sockaddr * pServerInfo;
+    uint16_t netPort = htons( port );
+    socklen_t serverInfoLength;
+
+    /* Perform a DNS lookup on the given host name. */
+    status = getaddrinfo( pServer, NULL, NULL, &pListHead );
+
+    if( status != -1 )
+    {
+        /* Attempt to connect to one of the retrieved DNS records. */
+        for( pIndex = pListHead; pIndex != NULL; pIndex = pIndex->ai_next )
+        {
+            tcpSocket = socket( pIndex->ai_family,
+                                pIndex->ai_socktype,
+                                pIndex->ai_protocol );
+
+            if( tcpSocket == -1 )
+            {
+                continue;
+            }
+
+            pServerInfo = pIndex->ai_addr;
+
+            if( pServerInfo->sa_family == AF_INET )
+            {
+                /* IPv4 */
+                ( ( struct sockaddr_in * ) pServerInfo )->sin_port = netPort;
+                serverInfoLength = sizeof( struct sockaddr_in );
+            }
+            else
+            {
+                /* IPv6 */
+                ( ( struct sockaddr_in6 * ) pServerInfo )->sin6_port = netPort;
+                serverInfoLength = sizeof( struct sockaddr_in6 );
+            }
+
+            status = connect( tcpSocket, pServerInfo, serverInfoLength );
+
+            if( status == -1 )
+            {
+                close( tcpSocket );
+            }
+            else
+            {
+                break;
+            }
+        }
+
+        if( pIndex == NULL )
+        {
+            /* Fail if no connection could be established. */
+            status = -1;
+        }
+        else
+        {
+            status = tcpSocket;
+        }
+    }
+
+    if( pListHead != NULL )
+    {
+        freeaddrinfo( pListHead );
+    }
+
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief The transport send function that defines the transport interface.
```

Comment:
It would be helpful for the customer to mention here that this would be used as the `HTTPTransportInterface.send` function to inject in the HTTP library in the demo,

Similarly, for the `transportRecv` function.